pyfar/classes/coordinates.py

```diff
@@ -919,24 +1237,133 @@ def find_nearest_sph(self, points_1, points_2, points_3, distance,
             >>> coords = pf.samplings.sph_lebedev(sh_order=10)
             >>> result = coords.find_nearest_sph(0, 0, 1, 45, show=True)
         """
+        warnings.warn((
+            "This function will be deprecated in pyfar 0.7.0 in favor "
+            "of find_nearest_by_distance."),
+                PendingDeprecationWarning)
 
         # check the input
         assert distance >= 0 and distance <= 180, \
             "distance must be >= 0 and <= 180."
 
-        # get radius and check for equality
-        radius = self.get_sph()[..., 2]
-        delta_radius = np.max(radius) - np.min(radius)
-        if delta_radius > 1e-15:
-            raise ValueError(
-                "find_nearest_sph only works if all points have the same \
-                radius. Differences are larger than 1e-15")
+        # get target point in cartesian coordinates
+        coords = Coordinates(
+            points_1, points_2, points_3, domain, convention, unit)
 
         # get the points
         distance, index, mask = self._find_nearest(
-            points_1, points_2, points_3,
-            domain, convention, unit, show, distance, 'sph', atol,
-            np.max(radius))
+            coords, show, distance, 'sph', atol)
+
+        return index, mask
+
+    def find_nearest_points(self, coords, number_of_points=1):
```

Comment:
replaces 
`find_nearest_k(self, points_1, points_2, points_3, k=1,
                       domain='cart', convention='right', unit='met',
                       show=False):`
what should be the return value, if we fix one we can do sth like 
`found = coords[coords.find_nearest(coords2find)]`
which would be very clean. I would also go for indexes then the `coords2find` Coordinates can be more than one and it can be used to restructure arrays. (same for slice)